packages-next/keystone/src/lib/core/ListTypes/list.ts

```diff
@@ -846,38 +918,68 @@ export class List implements BaseKeystoneList {
+    if (!access) {
+      // Return an access denied error for all items
+      return data.map(() => Promise.reject(new AccessDeniedError({ data: { type: 'mutation' } })));
+    }
+
     const existingItems = (await this.getAccessControlledItems(ids, access)) as Record<
       string,
       any
     >[];
 
     // Only update those items which pass access control
-    const itemsToUpdate = zipObj({
+    const _itemsToUpdate = zipObj({
       existingItem: existingItems,
       id: existingItems.map(({ id }) => id), // itemId is taken from here in checkFieldAccess
       data: existingItems.map(({ id }) => data.find(d => d.id === id)!.data),
     }) as { existingItem: Record<string, any>; id: IdType; data: Record<string, any> }[];
 
-    // FIXME: We should do all of these in parallel and return *all* the field access violations
-    await this.checkFieldAccess(operation, itemsToUpdate, context, extraData);
+    // Put items to update back into the same order as the original IDs
+    const itemsToUpdateById = Object.fromEntries(_itemsToUpdate.map(o => [o.id, o]));
+    const itemsToUpdate = ids.map(id => itemsToUpdateById[id] || null);
 
+    const fieldAccess = await this._returnFieldAccess(operation, itemsToUpdate, context, extraData);
+    let results: PromiseSettledResult<any>[] = [];
     if (this.adapter.parentAdapter.provider === 'sqlite') {
       // We perform these operations sequentially as a workaround for a connection
       // timeout bug that happens in prisma+sqlite: https://github.com/prisma/prisma/issues/2955
-      const ret = [];
+      let i = 0;
       for (const item of itemsToUpdate) {
-        const { existingItem, id, data } = item;
-        ret.push(await this._updateSingle(id, data, existingItem, context, mutationState));
+        if (item === null || fieldAccess[i] !== null) {
+          // The item either didn't exist, or was filtered out by access control
+          results.push({
+            status: 'rejected',
+            reason: new AccessDeniedError({ data: { type: 'mutation' } }),
+          });
+        } else {
+          const { existingItem, id, data } = item;
+          try {
+            const result = await this._updateSingle(id, data, existingItem, context, mutationState);
+            results.push({ status: 'fulfilled', value: result });
+          } catch (e) {
+            results.push({ status: 'rejected', reason: e });
+          }
+        }
+        i++;
       }
-      return ret;
     } else {
-      return Promise.all(
-        itemsToUpdate.map(({ existingItem, id, data }) =>
-          this._updateSingle(id, data, existingItem, context, mutationState)
-        )
+      results = await Promise.allSettled(
+        itemsToUpdate.map((item, i) => {
+          if (item === null || fieldAccess[i] !== null) {
+            // The item either didn't exist, or was filtered out by access control
+            return new AccessDeniedError({ data: { type: 'mutation' } });
```

Comment:
I know it presently works because graphql-js does magic things if you return an Error it seems but for clarity:
```suggestion
            return Promise.reject(new AccessDeniedError({ data: { type: 'mutation' } }));
```